File: openapi-test-kotlin-server-spring-client-mp/testapi/api/PetsApiClient.kt
package testapi.api

import com.fasterxml.jackson.databind.JsonNode
import io.swagger.v3.oas.annotations.security.SecurityRequirement
import jakarta.ws.rs.Consumes
import jakarta.ws.rs.DELETE
import jakarta.ws.rs.GET
import jakarta.ws.rs.POST
import jakarta.ws.rs.Path
import jakarta.ws.rs.Produces
import jakarta.ws.rs.WebApplicationException
import jakarta.ws.rs.core.MediaType
import jakarta.ws.rs.core.Response
import java.lang.IllegalStateException
import java.lang.Void
import java.util.Optional
import kotlin.collections.List
import testapi.model.Error
import testapi.model.Pet
import testapi.model.PetCreate
import testapi.model.PetId

interface PetsApiClient : PetsApi {
  /** Create a pet - handles response status codes */
  override fun createPet(body: PetCreate): Response201400<Pet, Error> {
    var response: Response
    try {
      response = createPetRaw(body);
    } catch (e: WebApplicationException) {
      response = e.getResponse();
    } 
    if (response.getStatus() == 201) { return Created(response.readEntity(Pet::class.java)) }
    else if (response.getStatus() == 400) { return BadRequest(response.readEntity(Error::class.java)) }
    else { throw IllegalStateException("Unexpected status code: " + response.getStatus()) }
  }

  /** Create a pet */
  @POST
  @Path("/")
  @Consumes(value = [MediaType.APPLICATION_JSON])
  @Produces(value = [MediaType.APPLICATION_JSON])
  @SecurityRequirement(name = "oauth2", scopes = ["write:pets"])
  @SecurityRequirement(name = "apiKeyHeader")
  fun createPetRaw(body: PetCreate): Response

  /** Delete a pet */
  @DELETE
  @Path("/{petId}")
  override fun deletePet(
    /** The pet ID */
    petId: PetId
  ): Void

  /** Get a pet by ID - handles response status codes */
  override fun getPet(
    /** The pet ID */
    petId: PetId
  ): Response200404<Pet, Error> {
    var response: Response
    try {
      response = getPetRaw(petId);
    } catch (e: WebApplicationException) {
      response = e.getResponse();
    } 
    if (response.getStatus() == 200) { return Ok(response.readEntity(Pet::class.java)) }
    else if (response.getStatus() == 404) { return NotFound(response.readEntity(Error::class.java)) }
    else { throw IllegalStateException("Unexpected status code: " + response.getStatus()) }
  }

  /** Get pet photo */
  @GET
  @Path("/{petId}/photo")
  @Produces(value = [MediaType.APPLICATION_OCTET_STREAM])
  override fun getPetPhoto(
    /** The pet ID */
    petId: PetId
  ): Void

  /** Get a pet by ID */
  @GET
  @Path("/{petId}")
  @Produces(value = [MediaType.APPLICATION_JSON])
  fun getPetRaw(
    /** The pet ID */
    petId: PetId
  ): Response

  /** List all pets */
  @GET
  @Path("/")
  @Produces(value = [MediaType.APPLICATION_JSON])
  override fun listPets(
    /** Maximum number of pets to return */
    limit: Optional<Integer>,
    /** Filter by status */
    status: Optional<String>
  ): List<Pet>

  /** Upload a pet photo */
  @POST
  @Path("/{petId}/photo")
  @Consumes(value = [MediaType.MULTIPART_FORM_DATA])
  @Produces(value = [MediaType.APPLICATION_JSON])
  override fun uploadPetPhoto(
    /** The pet ID */
    petId: PetId,
    /** Optional caption for the photo */
    caption: String,
    /** The photo file to upload */
    file: Array<Byte>
  ): JsonNode
}

File: openapi-test-scala-server-spring-client-jdk/testapi/api/PetsApiClient.scala
package testapi.api

import com.fasterxml.jackson.databind.ObjectMapper
import io.circe.Json
import java.lang.Exception
import java.net.URI
import java.net.http.HttpClient
import java.util.Optional
import testapi.model.Error
import testapi.model.Pet
import testapi.model.PetCreate
import testapi.model.PetId

/** JDK HTTP Client implementation for PetsApi */
class PetsApiClient(
  /** JDK HTTP client for making HTTP requests */
  val httpClient: HttpClient,
  /** Base URI for API requests */
  val baseUri: URI,
  /** Jackson ObjectMapper for JSON serialization */
  val objectMapper: ObjectMapper
) extends PetsApi {
  /** Create a pet */
  @throws[Exception]
  override def createPet(body: PetCreate): Response201400[Pet, Error] = {
    var request = java.net.http.HttpRequest.newBuilder(java.net.URI.create(baseUri.toString() + "/" + "pets")).method("POST", java.net.http.HttpRequest.BodyPublishers.ofString(objectMapper.writeValueAsString(body))).header("Content-Type", "application/json").header("Accept", "application/json").build()
    var response = httpClient.send(request, java.net.http.HttpResponse.BodyHandlers.ofString())
    var statusCode = response.statusCode()
    if (statusCode == 201) return new testapi.api.Created(objectMapper.readValue(response.body(), classOf[testapi.model.Pet]))
    else if (statusCode == 400) return new testapi.api.BadRequest(objectMapper.readValue(response.body(), classOf[testapi.model.Error]))
    else throw new java.lang.IllegalStateException(s"Unexpected status code: ${statusCode}")
  }

  /** Delete a pet */
  @throws[Exception]
  override def deletePet(
    /** The pet ID */
    petId: PetId
  ): Unit = {
    var request = java.net.http.HttpRequest.newBuilder(java.net.URI.create(baseUri.toString() + "/" + "pets" + "/" + petId.toString())).method("DELETE", java.net.http.HttpRequest.BodyPublishers.noBody()).header("Content-Type", "application/json").header("Accept", "application/json").build()
    var response = httpClient.send(request, java.net.http.HttpResponse.BodyHandlers.ofString())
  }

  /** Get a pet by ID */
  @throws[Exception]
  override def getPet(
    /** The pet ID */
    petId: PetId
  ): Response200404[Pet, Error] = {
    var request = java.net.http.HttpRequest.newBuilder(java.net.URI.create(baseUri.toString() + "/" + "pets" + "/" + petId.toString())).method("GET", java.net.http.HttpRequest.BodyPublishers.noBody()).header("Content-Type", "application/json").header("Accept", "application/json").build()
    var response = httpClient.send(request, java.net.http.HttpResponse.BodyHandlers.ofString())
    var statusCode = response.statusCode()
    if (statusCode == 200) return new testapi.api.Ok(objectMapper.readValue(response.body(), classOf[testapi.model.Pet]))
    else if (statusCode == 404) return new testapi.api.NotFound(objectMapper.readValue(response.body(), classOf[testapi.model.Error]))
    else throw new java.lang.IllegalStateException(s"Unexpected status code: ${statusCode}")
  }

  /** Get pet photo */
  @throws[Exception]
  override def getPetPhoto(
    /** The pet ID */
    petId: PetId
  ): Unit = {
    var request = java.net.http.HttpRequest.newBuilder(java.net.URI.create(baseUri.toString() + "/" + "pets" + "/" + petId.toString() + "/" + "photo")).method("GET", java.net.http.HttpRequest.BodyPublishers.noBody()).header("Content-Type", "application/json").header("Accept", "application/json").build()
    var response = httpClient.send(request, java.net.http.HttpResponse.BodyHandlers.ofString())
  }

  /** List all pets */
  @throws[Exception]
  override def listPets(
    /** Maximum number of pets to return */
    limit: Optional[Integer],
    /** Filter by status */
    status: Optional[String]
  ): java.util.List[Pet] = {
    var request = java.net.http.HttpRequest.newBuilder(java.net.URI.create(baseUri.toString() + "/" + "pets" + (if (limit.isPresent()) "?limit=" + limit.get().toString() else "") + (if (status.isPresent()) "&status=" + status.get().toString() else ""))).method("GET", java.net.http.HttpRequest.BodyPublishers.noBody()).header("Content-Type", "application/json").header("Accept", "application/json").build()
    var response = httpClient.send(request, java.net.http.HttpResponse.BodyHandlers.ofString())
    return objectMapper.readValue(response.body(), new com.fasterxml.jackson.core.`type`.TypeReference[java.util.List[testapi.model.Pet]] {})
  }

  /** Upload a pet photo */
  @throws[Exception]
  override def uploadPetPhoto(
    /** The pet ID */
    petId: PetId,
    /** Optional caption for the photo */
    caption: String,
    /** The photo file to upload */
    file: Array[Byte]
  ): Json = {
    var request = java.net.http.HttpRequest.newBuilder(java.net.URI.create(baseUri.toString() + "/" + "pets" + "/" + petId.toString() + "/" + "photo")).method("POST", java.net.http.HttpRequest.BodyPublishers.noBody()).header("Content-Type", "application/json").header("Accept", "application/json").build()
    var response = httpClient.send(request, java.net.http.HttpResponse.BodyHandlers.ofString())
    return objectMapper.readValue(response.body(), classOf[io.circe.Json])
  }
}